docs/admin/circuit-breaker.rst

```diff
@@ -0,0 +1,94 @@
+.. _circuit-breaker:
+
+===============
+Circuit Breaker
+===============
+CrateDB's circuit breaker prevents queries and background processes from exhausting node memory and destabilizing the cluster.
+It works by estimating the memory an operation will require and aborting the operation before the JVM heap space is exhausted.
+
+What is a Circuit Breaker
+==========================
+A **circuit breaker** is a safeguard that halts work when resource usage crosses a dangerous threshold.
+Think of the miniature breakers inside a household fuse box: if too many appliances pull current on the same line, the breaker
+trips and cuts power to prevent the wires from melting. The same principle applies in software, only the resource under pressure
+is memory, CPU, file descriptors, or an external service.
+
+In CrateDB, the critical resource is **RAM**. Queries run in parallel across many shards; a single 
+oversized aggregation or JOIN can allocate gigabytes in milliseconds. The breaker detects this and aborts the query with a 
+``CircuitBreakingException`` instead of letting the JVM run out of heap and crash the node.
+
+How Circuit Breakers Work in CrateDB
+====================================
+A query executes as an ordered set of operations. Before running each stage, CrateDB estimates the extra memory that step will need.
+If the projected total would exceed the breaker limit, the system aborts the query and returns a ``CircuitBreakingException``. 
+This pre-emptive trip prevents the JVM's garbage collector from reaching an unrecoverable out-of-memory state.
+
+Types of Circuit Breakers
+=========================
+There are five different Circuit Breaker types which are described in detail in the `cluster settings`_ documentation page: ``query``,
+``request``, ``jobs_log``, ``operations_log``, ``total`` and ``accounting``, which was deprecated and will be removed soon. The last one, which is also 
+known as ``parent`` circuit breaker, accounts for all others, meaning that it controls the general use of memory, tripping an operation if a 
+combination of the circuit breakers threatens the cluster. 
+
+Monitoring & Observability
+==========================
+
+To monitor the Circuit Breaker usage, follow the `JMX monitoring`_ guide. In particular,
+consult the `CircuitBreakers MXBean`_ section.
+
+For hosted deployments, see `Cloud monitoring`_ and for self-managed clusters, the `on-prem monitoring`_ guide. Both describe the complete path from cluster
+deployment to collecting metrics and displaying them on a Grafana dashboard.
+
+Exception Handling
+==================
+.. code-block:: console
+  
+  CircuitBreakingException: [query] Data too large, data for [mergeOnHandler: 1] would be [9021423616/8.4gb], which is larger than the limit of [9019431321/8.3gb]
```

Comment:
I think this is an outdated version of the error message. It was changed recently to remove the double numbers (`[9021423616/8.4gb]`) and hints what to do were added directly the the error.